lib/sinon/spy-entire-object.js

```diff
@@ -0,0 +1,22 @@
+"use strict";
+
+var getPropertyDescriptor = require("./util/core/get-property-descriptor");
+var walk = require("./util/core/walk");
+
+function spyEntireObject(spy, object) {
+    walk(object || {}, function(prop, propOwner) {
+        // we don't want to spy things like toString(), valueOf(), etc. so we only spy if the object
+        // is not Object.prototype
+        if (
+            propOwner !== Object.prototype &&
+            prop !== "constructor" &&
```

Comment:
We may also want to spy on the `constructor` for the same reasons above. I think reducing edge-cases would be productive here.

Reply:
This is a blatant copy of stub-entire-object.js as I wanted to keep the API for the two in sync.

Reply:
FYI, I also agree with the way it was originally written. Spying on native object methods seems out of scope...

Reply:
I see. That's fine then, I didn't know that, sorry 😅